core/src/main/java/fr/sncf/osrd/standalone_sim/result/ElectrificationConditionsRange.java

```diff
@@ -0,0 +1,66 @@
+package fr.sncf.osrd.standalone_sim.result;
+
+import static fr.sncf.osrd.envelope_sim.EnvelopeSimPath.ElectrificationConditions;
+
+import com.google.common.collect.RangeMap;
+import com.squareup.moshi.Json;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ElectrificationConditionsRange {
+    public final double start;
+    public final double stop;
+    @Json(name = "used_mode")
+    public final String usedMode;
+    @Json(name = "used_profile")
+    public final String usedProfile;
+    @Json(name = "used_restriction")
+    public final String usedRestriction;
+    @Json(name = "seen_mode")
+    public String seenMode = null;
+    @Json(name = "seen_profile")
+    public String seenProfile = null;
+    @Json(name = "seen_restriction")
+    public String seenRestriction = null;
+
+    /**
+     * ElectrificationConditionsRange constructor
+     */
+    public ElectrificationConditionsRange(double start, double stop, ElectrificationConditions usedCond,
+                                          ElectrificationConditions seenCond) {
+        this.start = start;
+        this.stop = stop;
+        this.usedMode = usedCond.mode();
+        this.usedProfile = usedCond.profile();
+        this.usedRestriction = usedCond.powerRestriction();
+        if (seenCond != null) {
+            this.seenMode = this.usedMode == null || !this.usedMode.equals(seenCond.mode()) ? seenCond.mode() : null;
```

Comment:
please don't use ternaries, I'd rather read longer but easier to understand code
```suggestion
            this.seenMode = null;
            if (this.usedMode == null || !this.usedMode.equals(seenCond.mode()))
                this.seenMode = seenCond.mode();
```